.gitignore

```diff
@@ -19,9 +19,9 @@
 .DS_Store
 
 # CMake artifacts
-build/
-build-*/
-Testing/
+/build/
+/build-*/
+/Testing/
```

Comment:
I have `compiler/build/` and `runtime/build/` directories from I think Python package builds that would be un-ignored by this change. Could either
* Give the new `compiler/bindings/python/iree/build/` package a different name ("builder"?)
* Explicitly _include_ the `compiler/iree/build/` directory:
    ```gitignore
    
    # CMake artifacts
    build/
    build-*/
    Testing/
    
    # Include iree.build package
    !compiler/bindings/python/iree/compiler/build/
    ```
* Explicitly _exclude_ those directories
    ```gitignore
    # CMake artifacts
    /build/
    /build-*/
    /Testing/
    /compiler/build/
    /runtime/build/
    ```

Oh, and I also have a few others from Python too:
* `integrations/tensorflow/python_projects/iree_tf/build/lib/iree/tools/tf/scripts/iree_import_tf/__main__.py`
* `integrations/tensorflow/python_projects/iree_tflite/build/lib/iree/tools/tflite/__init__.py`